Rework models and crawl responses

tw4l opened the pull request:
Fixes #981 

The following fields are now added in the GET /all-crawl endpoints (single crawl and list) for automated crawls only:

- cid
- name
- description
- firstSeed
- seedCount
- profileName

This provides sufficient data for the frontend's All Archived Data tab to be consistent between All and Crawls.

Since it's not possible to add additional aggregation stages within a `cond` in mongo, these fields are added by annotating the items after the aggregation completes in the list endpoint.

In the process of getting there:
- All pydantic models have been moved to a new `models.py` module, and de-duplicated in some cases (there may be more room here to continue on that front, but I didn't want the scope to get too out of control).
- The `PaginatedResponse` model has been added as the response_model for all list GET endpoints
- Some of the get_<resource> methods that return a single resource were reworked to optionally include or exclude resources
- Removed `files` from crawl output model: only return `resources` which conform with frictionless data format and include presigned URLs.

tw4l committed:
Include automated crawl details in all-crawls GET endpoints

Fields added in get and list all-crawl endpoints for automated
crawls only:

- cid
- name
- description
- firstSeed
- seedCount
- profileName

This provides sufficient data for the frontend All Archived Data tab
to be consistent between All and Crawl tabs

tw4l committed:
Add automated crawl fields to list all-crawls test

ikreymer commented on backend/btrixcloud/basecrawls.py:
```diff
@@ -258,6 +272,57 @@ async def _delete_crawl_files(self, crawl, org: Organization):
+
+        # if running, get stats directly from redis
+        # more responsive, saves db update in operator
+        if crawl.state in (
```
We should move the list of running states to basecrawls to avoid duplications / possible errors if we add more states. Should we just move _resolve_crawl_refs() to be here and avoid further duplication?

tw4l replied:
Yup, good points! I'll move both to basecrawls

tw4l committed:
WIP: Move _resolve_crawl_refs and states to basecrawls

tw4l committed:
Move crawl-related models to models module

tw4l committed:
Add upload pydantic models to models.py and deduplicate models

tw4l committed:
Fix final failing test

tw4l commented:
@ikreymer This is ready for re-review. To make it work I ended up moving all of the various crawl + crawl config pydantic model definitions into a new, separate module - lmk what you think.

ikreymer commented:
> @ikreymer This is ready for re-review. To make it work I ended up moving all of the various crawl + crawl config pydantic model definitions into a new, separate module - lmk what you think.

Nice! yes, i was thinking we'd need a models module eventually.. should we move all model definitions there while we're at it? or as a follow up PR perhaps?

tw4l commented:
> Nice! yes, i was thinking we'd need a models module eventually.. should we move all model definitions there while we're at it? or as a follow up PR perhaps?

I'm happy to continue with that in this PR and get it all done at once :) Will ping again when it's done!

tw4l committed:
Move collections models to models.py

tw4l committed:
Move invites models to models.py

tw4l committed:
Move Org models to models.py

tw4l committed:
Move PaginatedResponse to models.py

tw4l committed:
Move profile models to models.py

tw4l committed:
Reformat pagination

tw4l committed:
Remove BaseModel import from pagination

tw4l committed:
Move User models to models.py

tw4l committed:
Move User model up in models.py

tw4l committed:
Fix order of user models in models.py

tw4l committed:
Fix cyclic import

tw4l committed:
Remove unused import

tw4l committed:
Add PaginatedResponse model to all list endpoints

tw4l committed:
Uncomment mongo readinessProbe

tw4l commented:
@ikreymer Done! I also tested locally in the frontend to make sure all looked good there and didn't spot any problems

tw4l committed:
Add pylint disable comments back

ikreymer committed:
cleanup CrawlOutWithResources:
- remove 'files' from output model, only resources should be returned
- add _files_to_resources() to simplify computing presigned 'resources' from raw 'files'
- update upload tests to be more consistent, 'files' never present, 'errors' always none

ikreymer committed:
format fix
